lncfg/monitoring_off.go

```diff
@@ -0,0 +1,9 @@
+// !build monitoring
+
+package lncfg
+
+// PrometheusConfig configures the Prometheus exporter when monitoring is
+// enabled. Monitoring is currently disabled.
+type PrometheusConfig struct {
+	ListenAddr string `long:"listenaddr" description:"the interface we should listen on for prometheus"`
```

Comment:
If this is the version with prometheus off, then why do we need to expose this `ListenAddr` field?

Reply:
i think because [this](https://github.com/lightningnetwork/lnd/pull/3134/files#diff-01ada5680aaa18fc79cbf78b2d55a37bR607) will fail to compile

@valentinewallace we can get around this by defining a function, which when `_on` returns the `ListenAddr` and when `_off` returns `""`. that will allow us to vary the fields without breaking compilation

fwiw, most CLI args take `listen` as the arg as opposed to `listenaddr`

Reply:
Solved by passing the entire Prometheus config to `ExportPrometheusMetrics`.